### PR TITLE
FAI-491: Prepare background for run cypress test suite with real data

### PR DESCRIPTION
**Pull request by sjamboro**
FAI-491: Prepare background for run cypress test suite with real data
https://issues.redhat.com/browse/FAI-491

This PR define two test suites of Trusty Console:
**Simple ui unit tests** run with mocked data and they are in 'integration' folder.
They should contain quick check of simple user interaction based on
mocked data. To run mocked smoke test, you need to perform `yarn run test:it`

**E2E tests** run with real data. They are in 'e2e' folder. They require build simple kogito projects with DMN and
PMML assets. To run E2E smoke test, you need to start Trusty Console with kogito projects manually before you perform `yarn run cypress:run:e2e`

### CODE REVIEW

**Comment by sjamboro** [comment]
jenkins retest this

File: ui-packages/packages/trusty/package.json [hunk 1]
```diff
@@ -29,10 +29,12 @@
     "test:watch": "jest --watch",
     "lint": "eslint './src/**/*.ts{,x}'",
     "mock-server": "json-server --watch api-mock/db.js --routes api-mock/routes.json --port 1336 --delay 1000 --id executionId --middlewares api-mock/filterSingular.js",
-    "cypress:open": "yarn run cypress open --project cypress",
-    "cypress:run": "yarn run cypress run -b chrome --project cypress",
+    "cypress:open:unit": "yarn run cypress open --project cypress --config-file cypress.unit.json",
```
Comment (jstastny-cz) [inline ui-packages/packages/trusty/package.json]: Cypress tests are not unit-level IMO. Probably `it` vs `e2e` might be a better distinction.

Comment (jstastny-cz) [reply ui-packages/packages/trusty/package.json]: It will also match when we have `test:it` and `test:e2e` scripts.

**Review by jstastny-cz — COMMENTED** [review]
One thing I am struggling with - this is not complete configuration, wondering if it makes sense to merge as is. There's the part of starting the service itself missing. Couldn't this be done together?

File: ui-packages/packages/trusty/package.json (hunk 1 above)
Comment (sjamboro) [reply ui-packages/packages/trusty/package.json]: @jstastny-cz I am running service manually (FAI-507), right now.

Comment (jstastny-cz) [reply ui-packages/packages/trusty/package.json]: Ok, should be resolved by complementary #868 , fair enough.

File: ui-packages/packages/trusty/package.json [hunk 2]
```diff
@@ -29,12 +29,12 @@
...
-    "cypress:open:unit": "yarn run cypress open --project cypress --config-file cypress.unit.json",
-    "cypress:run:unit": "yarn run cypress run -b chrome --project cypress --config-file cypress.unit.json",
+    "cypress:open:it": "yarn run cypress open --project it-tests --config-file cypress.it.json",
+    "cypress:run:it": "yarn run cypress run -b chrome --project it-tests --config-file cypress.it.json",
     "start-and-mock": "yarn run start & yarn run mock-server",
-    "test:it": "start-server-and-test start-and-mock http://0.0.0.0:9000 cypress:run:unit",
-    "cypress:open:e2e": "yarn run cypress open --project cypress --config-file cypress.e2e.json",
-    "cypress:run:e2e": "yarn run cypress run -b chrome --project cypress --config-file cypress.e2e.json"
+    "test:it": "start-server-and-test start-and-mock http://0.0.0.0:9000 cypress:run:it",
+    "cypress:open:e2e": "yarn run cypress open --project it-tests --config-file cypress.e2e.json",
```
Comment (jstastny-cz) [inline ui-packages/packages/trusty/package.json:36]: should it be `--project e2e` here?

Comment (jstastny-cz) [reply ui-packages/packages/trusty/package.json:36]: no, it should not! :-)

Comment (jstastny-cz) [reply ui-packages/packages/trusty/package.json:36]: disregard.

**Review by jstastny-cz — APPROVED** [review]
LGTM.

**Comment by sjamboro** [comment]
@jstastny-cz this is a complementary PR. The backend is started manually. See https://issues.redhat.com/browse/FAI-507 and https://issues.redhat.com/browse/FAI-490.

File: ui-packages/packages/trusty/it-tests/e2e/smoke/SmokeTest.ts [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    cy.ouiaId('search-input').type('someId');
+    cy.ouiaId('search-button').should('be.visible');
+  });
+
+  it('Refresh is visible', () => {
+    cy.ouiaId('refresh-button').should('be.visible');
+  });
+
+  it('Date inputs are visible', () => {
+    cy.get('div.flatpickr-wrapper>input#audit-from-date+input').should(
```
Comment (kelvah) [inline ui-packages/packages/trusty/it-tests/e2e/smoke/SmokeTest.ts]: Why this selector is so specific? Isn't `#audit-from-date+input` enough? Same thing for the `to-date` field possibly.

**Review by kelvah — APPROVED** [review]
LGTM. Left you a comment.

File: ui-packages/packages/trusty/it-tests/e2e/smoke/SmokeTest.ts (hunk 3 above)
Comment (sjamboro) [reply ui-packages/packages/trusty/it-tests/e2e/smoke/SmokeTest.ts]: @kelvah you are right. It was copied from FAI-471. I tried to find a locator and I did not recognize that I decided to use id ("#").  Thank you, I will fix it. My target is to use ouia-id attribut but I have not found the solution for Data Picker element from PatternFly. Do you have some tips how I am able to add ouia-id for this element?

Comment (kelvah) [reply ui-packages/packages/trusty/it-tests/e2e/smoke/SmokeTest.ts]: @sjamboro We don't use PF datepickers in Trusty yet. At the time of development they didn't exist. So I relied on another library (flatpickr) as a temporary solution. I think that for now you can't avoid selecting both the field id and the sibling `input` tag, like you did.

**Comment by sjamboro** [comment]
@danielezonca can you merge this PR?

**Comment by manstis** [comment]
Is the test failure on Github Actions of concern?
```
2021-06-09T12:17:43.4940693Z [INFO] @kogito-apps/task-console: Summary of all failing tests
2021-06-09T12:17:43.4941948Z [INFO] @kogito-apps/task-console: FAIL src/components/Organisms/TaskInbox/tests/TaskInbox.test.tsx (15.203s)
2021-06-09T12:17:43.4943219Z [INFO] @kogito-apps/task-console:   ● TaskInbox tests › test sorting -> with direction
2021-06-09T12:17:43.4944443Z [INFO] @kogito-apps/task-console:     Method “props” is meant to be run on 1 node. 0 found instead.
2021-06-09T12:17:43.5037130Z [INFO] @kogito-apps/task-console:       483 |     await wait(20);
2021-06-09T12:17:43.5038475Z [INFO] @kogito-apps/task-console:       484 |     wrapper = wrapper.update();
2021-06-09T12:17:43.5039860Z [INFO] @kogito-apps/task-console:     > 485 |     expect(wrapper.find(DataTable).props()['sortBy']).toEqual({
2021-06-09T12:17:43.5041038Z [INFO] @kogito-apps/task-console:           |                                    ^
2021-06-09T12:17:43.5041954Z [INFO] @kogito-apps/task-console:       486 |       index: 3,
2021-06-09T12:17:43.5042810Z [INFO] @kogito-apps/task-console:       487 |       direction: 'asc'
2021-06-09T12:17:43.5043630Z [INFO] @kogito-apps/task-console:       488 |     });
2021-06-09T12:17:43.5044766Z [INFO] @kogito-apps/task-console:       at ReactWrapper.single (../../node_modules/enzyme/src/ReactWrapper.js:1168:13)
2021-06-09T12:17:43.5046324Z [INFO] @kogito-apps/task-console:       at ReactWrapper.props (../../node_modules/enzyme/src/ReactWrapper.js:710:17)
2021-06-09T12:17:43.5047715Z [INFO] @kogito-apps/task-console:       at src/components/Organisms/TaskInbox/tests/TaskInbox.test.tsx:485:36
2021-06-09T12:17:43.5049164Z [INFO] @kogito-apps/task-console:       at step (src/components/Organisms/TaskInbox/tests/TaskInbox.test.tsx:44:23)
2021-06-09T12:17:43.5050611Z [INFO] @kogito-apps/task-console:       at Object.next (src/components/Organisms/TaskInbox/tests/TaskInbox.test.tsx:25:53)
2021-06-09T12:17:43.5052163Z [INFO] @kogito-apps/task-console:       at fulfilled (src/components/Organisms/TaskInbox/tests/TaskInbox.test.tsx:16:58)
2021-06-09T12:17:43.5053374Z [INFO] @kogito-apps/task-console: Test Suites: 1 failed, 25 passed, 26 total
2021-06-09T12:17:43.5054408Z [INFO] @kogito-apps/task-console: Tests:       1 failed, 97 passed, 98 total
2021-06-09T12:17:43.5055454Z [INFO] @kogito-apps/task-console: Snapshots:   65 passed, 65 total
2021-06-09T12:17:43.5056502Z [INFO] @kogito-apps/task-console: Time:        39.157s
2021-06-09T12:17:43.5057388Z [INFO] @kogito-apps/task-console: Ran all test suites.
2021-06-09T12:17:43.5058929Z [INFO] @kogito-apps/management-console: PASS src/components/Molecules/ProcessListToolbar/tests/ProcessListToolbar.test.tsx
2021-06-09T12:17:43.5060809Z [INFO] @kogito-apps/task-console: error Command failed with exit code 1.
2021-06-09T12:17:43.5062311Z [INFO] @kogito-apps/task-console: info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
2021-06-09T12:17:43.5063727Z [INFO] @kogito-apps/task-console: error Command failed with exit code 1.
2021-06-09T12:17:43.5065081Z [INFO] @kogito-apps/task-console: info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
2021-06-09T12:17:43.5066616Z [INFO] lerna ERR! yarn run test:coverage exited 1 in '@kogito-apps/task-console'
```
It might be worth starting the GitHub Action build again?

**Comment by sjamboro** [comment]
@manstis I am going to run it again. Error is from another module @kogito-apps/task-console. I am going to put this branch on top of master.

**Comment by sjamboro** [comment]
Jenkins retest this

**Comment by sjamboro** [comment]
jenkins retest this

**Comment by manstis** [comment]
Jenkins retest this

**Comment by sjamboro** [comment]
jenkins retest this

**Comment by sjamboro** [comment]
@danielezonca can you merge this PR?

**Comment by sjamboro** [comment]
@manstis can you merge this PR?

**Pull request merged**